### PR TITLE
Preparing for the beta 3 release

### PR DESCRIPTION
**Pull request by rprouse**
**Ready to merge**

### CODE REVIEW

File: CHANGES.txt [hunk 1]
```diff
@@ -1,4 +1,59 @@
...
+ * 670 Failing Tests in TeamCity Build
+ * 673 Ensure proper disposal of engine objects
+ * 674 Engine does not release test assemblies
+ * 679 Windows 10 Support
+ * 682 Add Async Support to Portable Framework
+ * 683 Make FrameworkController available in portable build
+ * 687 TestAgency does not launch agent process correctly if runtime type is not specified (i.e. v4.0)
+ * 692 PlatformAttribute_OperatingSystemBitNess fails when running in 32-bit process
+ * 693 Generic Test<T> Method cannot determine type arguments for fixture when passed as IEnumerable<T>
+ * 698 Deprecate / Remove TestCaseSourceAttribute Constructors
```
Comment (CharliePoole) [inline CHANGES.txt]: I changed the title of this issue to "Require TestCaseSource and ValueSource named members to be static"

File: CHANGES.txt [hunk 2]
```diff
@@ -1,4 +1,59 @@
...
+
+Framework
+
+ * The RangeAttribute has been extended to support more data types including 
+   uint, long and ulong
+ * Added platform support for Windows 10 and fixed issues with Windows 8 and
+   8.1 support
+ * Added async support to the portable version of NUnit Framework
+ * Several TestCaseSourceAttribute constructors have been deprecated, see the
+   documentation for more information
```
Comment (CharliePoole) [inline CHANGES.txt]: We decided not to do this and instead require those constructors to be static. I forgot to change the title of the issue, but just did. Also applies to ValueSourceAttribute.

File: CHANGES.txt (hunk 1 above)
Comment (rprouse) [inline CHANGES.txt]: I saw the email and had already pushed the change :smile:

File: CHANGES.txt [hunk 3]
```diff
@@ -1,4 +1,59 @@
...
+ * RandomAttribute has been extended to add support for new data types including 
+   uint, long, ulong, short, ushort, float, byte and sbyte
```
Comment (CharliePoole) [inline CHANGES.txt]: Also TestContext.Random gets the same added support, plus decimal. I'd make that a separate bullet point.

File: CHANGES.txt (hunk 2 above)
Comment (rprouse) [inline CHANGES.txt]: I will update the description.

File: CHANGES.txt [hunk 4]
```diff
@@ -1,4 +1,59 @@
...
+ * Removed the dependency on Microsoft.Bcl.Async from the NUnit Framework assembly
+   targeting .NET 4.0. If you want to write async tests in .NET 4.0, you will need
+   to reference the NuGet package yourself.
```
Comment (CharliePoole) [inline CHANGES.txt]: Possible addition: Significant improvements have been made in how NUnit deduces the type arguments of generic methods based on the arguments provided.

Does this feel like a "feature" to you? Do we want users to know that this has substantially changed?

Comment (rprouse) [inline CHANGES.txt]: Sounds good, added. I have pushed changes based on all your comments.

**Comment by rprouse** [comment]
I've merged in master and will start building beta 3 based on this branch. If anyone sees anything else that needs updating, let me know :smile:

**Pull request merged**